src/lib/utils.py

```diff
@@ -68,6 +68,15 @@ def _extract_timestamp(info):
     raise CheckerRemoteError(f"Cannot parse date/time: {date_str}")
 
 
+def _check_newline(fp):
+    fp.seek(0, 2)
+    fp.seek(fp.tell() - 1, 0)
```

Comment:
Please use named constants instead of numeric values, e.g. `os.SEEK_END` instead of `2`.
Also I guess this can be done in one action with negative offset instead of two, i.e. `fp.seek(-1, os.SEEK_END)`?

Reply:
> Please use named constants instead of numeric values, e.g. os.SEEK_END instead of 2

Will do.

> Also I guess this can be done in one action with negative offset instead of two, i.e. fp.seek(-1, os.SEEK_END)?

Technically yes, but it requires opening the file in binary mode and I was afraid of edge cases with different encodings. Should I change it?

Reply:
Right. I'm not sure if `json.dump` will play nice with binary IO, so it's probably safer to leave it as is, then.